### PR TITLE
Migrate to Github Actions

### LINKED ISSUE #12: Move build to Github Actions
We currently use Travis CI to build (& release) the .jar-file.

We should move this to Github Actions as this is better integrated into Github.

### PR DESCRIPTION
**Pull request by kloessst**
### Checklist
- [x] I built and deployed the plugin locally and tested *all* changes and *all* related features.

### Motivation and Context

Migrate from Travis to GitHub Actions, as Travis was deactivated for this organization.
Closes https://github.com/ls1intum/bamboo-server-notification-plugin/issues/12

### Description
Added Spotless to project.

Migrate from Travis to GitHub Actions. Three workflows are defined:
- On PR and commit to master: Checks the format, builds, tests and uploads the package.
- The following steps are executed additionally on tag creation: create release, upload release artifact

### Steps for Testing
1. Review the workflows

### CODE REVIEW

File: src/main/java/de/tum/in/www1/bamboo/server/ServerNotificationRecipient.java [hunk 1]
```diff
@@ -1,82 +1,80 @@
...
     /*
-     * The TestResults of successful tests can not be loaded from the ChainResultSummary (only failing ones can).
-     * The TestResults need a BuildContext and can therefor only be accessed when using an Event that extends BuildContextEvent.
-     * The normal BuildCompletedEvent does not extend BuildContextEvent, but the PostBuildCompletedEvent does.
-     * We listen for the PostBuildCompletedEvent and save the test cases in this Map with the key of the job as String.
-     * The TestResults are stored inside the ResultsContainer class and it can be retrieved using this Map in the ServerNotificationTransport class.
-     * A method clearOldTestResultsContainer() has been added, that removes old ResultsContainer from the Map, because we add every build to this Map, even those without
-     * Notifications enabled.
-     * The time (in seconds) after the ResultsContainer can be specified in the variable TESTRESULTSCONTAINER_REMOVE_TIME.
+     * The TestResults of successful tests can not be loaded from the ChainResultSummary (only failing ones can). The TestResults need a BuildContext and can therefor only be
```
Comment (sleiss) [inline src/main/java/de/tum/in/www1/bamboo/server/ServerNotificationRecipient.java]: I actually preferred the old formatting here. Can we disable the formatter for this part?

File: src/main/java/de/tum/in/www1/bamboo/server/ResultsContainer.java [hunk 2]
```diff
@@ -59,13 +60,7 @@ public long getInitTimestamp() {
 
     @Override
     public String toString() {
-        return "ResultsContainer{" +
```
Comment (sleiss) [inline src/main/java/de/tum/in/www1/bamboo/server/ResultsContainer.java]: I actually preferred the old formatting here. Can we disable the formatter for this part?

File: src/main/java/de/tum/in/www1/bamboo/server/ServerNotificationRecipient.java (hunk 1 above)
Comment (kloessst) [reply src/main/java/de/tum/in/www1/bamboo/server/ServerNotificationRecipient.java]: I have to look into the options file.

File: src/main/java/de/tum/in/www1/bamboo/server/ResultsContainer.java (hunk 2 above)
Comment (kloessst) [reply src/main/java/de/tum/in/www1/bamboo/server/ResultsContainer.java]: I have to look into the options file.

File: src/main/java/de/tum/in/www1/bamboo/server/ServerNotificationRecipient.java (hunk 1 above)
Comment (kloessst) [reply src/main/java/de/tum/in/www1/bamboo/server/ServerNotificationRecipient.java]: Deactivated the option for formatting of block comments and returned to the old state.

File: src/main/java/de/tum/in/www1/bamboo/server/ResultsContainer.java (hunk 2 above)
Comment (kloessst) [reply src/main/java/de/tum/in/www1/bamboo/server/ResultsContainer.java]: Found the option

**Review by sleiss — APPROVED** [review]
Built the plugin locally and deployed it, basic functionality still works. Workflows look good to me, however I'm not really experienced with them. As it seemed to work on @kloessst's branch of this plugin, it should be fine.

**Review by krusche — APPROVED** [review]

**Pull request merged**